homework-2/src/components/features/shows/ShowContainer/ShowContainer.tsx

```diff
@@ -1,28 +1,91 @@
 import useSWR from "swr"
-import { getShowDetails } from "@/fetchers/shows"
 import { useParams } from "next/navigation"
+import { swrKeys } from "@/fetchers/swrKeys"
+import { fetcher } from "@/fetchers/fetcher"
+import { IShow } from "@/typings/Show.type"
+import { IReview } from "@/typings/Review.type"
 
 export const ShowContainer = () => {
     const [averageRating, setAverageRating] = useState(0);
+    const [reviewsList, setReviewsList] = useState<IReview[]>([]);
 
     const params = useParams();
 
-    const { data, error, isLoading } = useSWR(`/shows/${params.id}`, () => getShowDetails(params.id as string));
+    const id = params.id as string;
 
-    if (isLoading) return <div>loading...</div>
+    const { data, error, isLoading } = useSWR<IShow>(swrKeys.showdetails(id), fetcher);
+    
+    useEffect(() => {
+        console.log("useEffect triggered");
+        const loadedFromLS = loadFromLocalStorage(id);
+
+        setAverageRating(calcAvgRating(loadedFromLS));
+        setReviewsList(loadedFromLS);
+        setAverageRating(calcAvgRating(loadedFromLS));
+    }, []);
 
+    useEffect(() => {
+        console.log("useSWR data:", data);  
+    }, [data]);
+        
+    const saveToLocalStorage = (reviewsList: IReview[], id: string) => {
+        if(reviewsList.length > 0){
+            localStorage.setItem(`review-list-${id}`, JSON.stringify(reviewsList));
+        } else {
+            localStorage.removeItem(`review-list-${id}`);
+        }
+    };
+    
+    const loadFromLocalStorage = (id: string) => {
+        const reviewListString = localStorage.getItem(`review-list-${id}`);
+        return reviewListString ? JSON.parse(reviewListString) : [];
+    }
+    
+    const onAddReview = (review: IReview) => {
+        const newList = [...reviewsList, review];
+        
+        setReviewsList(newList);
+        saveToLocalStorage(newList, id);
+        setAverageRating(calcAvgRating(newList));
+    }
+    
+    const onDeleteReview = (reviewToRemove: IReview) => {
+        const newList = reviewsList.filter((review) => review !== reviewToRemove);
+        
+        setReviewsList(newList);
+        saveToLocalStorage(newList, id);
+        setAverageRating(calcAvgRating(newList));
+    }
+    
+    let calcAvgRating = (reviews: IReview[]) => {
+        let sum = 0;
+        
+        reviews.forEach((review) => {
+            sum += review.rating;
+        });
+        
+        return Number((sum / reviews.length).toFixed(2));
+    }
+    
+    if (isLoading) return <div>loading...</div>
+    
     if (error) return <div>failed to load</div>
 
+    if(!isLoading){
+        if(!data)
+            return null;    
+    }
+
     return (
         <Flex flexDirection='column' alignItems='left' width='920px'>
-            <ShowDetails show={data} avgRating={averageRating} />
-            <ShowReviewSection setAvgRating={setAverageRating} />
+            {data && <ShowDetails show={data} avgRating={averageRating} />}

```

Comment:
Ovdje moraš prosljeđivati `data.show`
```suggestion
            {data && <ShowDetails show={data.show} avgRating={averageRating} />}
```

Reply:
Davao je error kad sam pokushao da uradim ovo. Sad je rjesheno 😃 🤦🏻‍♂️